James Louis  
Department of Engineering and Technology, Northcentral University  
DDS8555: Predictive Analysis  
Dr. Sebhat Gezehey  
December 20, 2025  


In [10]:
#import libaries
import pandas as pd
import statsmodels.formula.api as smf

In [2]:
# Load the Carseats dataset
carseats = pd.read_csv("Carseats.csv")

# Preview the data
carseats.head()


Sales  CompPrice  Income  Advertising  Population  Price ShelveLoc  Age  \
0   9.50        138      73           11         276    120       Bad   42   
1  11.22        111      48           16         260     83      Good   65   
2  10.06        113      35           10         269     80    Medium   59   
3   7.40        117     100            4         466     97    Medium   55   
4   4.15        141      64            3         340    128       Bad   38   

   Education Urban   US  
0         17   Yes  Yes  
1         10   Yes  Yes  
2         12   Yes  Yes  
3         14   Yes  Yes  
4         13   Yes   No

In [3]:
#Code Structure & Summary

print(carseats.info())
print(carseats.describe())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sales        400 non-null    float64
 1   CompPrice    400 non-null    int64  
 2   Income       400 non-null    int64  
 3   Advertising  400 non-null    int64  
 4   Population   400 non-null    int64  
 5   Price        400 non-null    int64  
 6   ShelveLoc    400 non-null    object 
 7   Age          400 non-null    int64  
 8   Education    400 non-null    int64  
 9   Urban        400 non-null    object 
 10  US           400 non-null    object 
dtypes: float64(1), int64(7), object(3)
memory usage: 34.5+ KB
None
            Sales   CompPrice      Income  Advertising  Population  \
count  400.000000  400.000000  400.000000   400.000000  400.000000   
mean     7.496325  124.975000   68.657500     6.635000  264.840000   
std      2.824115   15.334512   27.986037     6.650364  147.376436   
m

In [4]:
# Convert qualitative variables so statsmodels handles them correctly
carseats["Urban"] = carseats["Urban"].astype("category")
carseats["US"] = carseats["US"].astype("category")


In [8]:
# (a) Fit a multiple regression model to predict Sales using Price, Urban, and US
model_full = smf.ols("Sales ~ Price + Urban + US", data=carseats).fit()
model_full.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.239
Model:                            OLS   Adj. R-squared:                  0.234
Method:                 Least Squares   F-statistic:                     41.52
Date:                Sun, 21 Dec 2025   Prob (F-statistic):           2.39e-23
Time:                        09:47:12   Log-Likelihood:                -927.66
No. Observations:                 400   AIC:                             1863.
Df Residuals:                     396   BIC:                             1879.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept       13.0435      0.651     20.036      0.000      11.764      14.323
Urban[T.Yes]    -0.0219      0.272     -0.081      0.936      -0.556       0.512
US[T.Yes]        1.2006      0.259      4.635      0.000       0.691       1.710
Price           -0.0545      0.005    -10.389      0.000      -0.065      -0.044
==============================================================================
Omnibus:                        0.676   Durbin-Watson:                   1.912
Prob(Omnibus):                  0.713   Jarque-Bera (JB):                0.758
Skew:                           0.093   Prob(JB):                        0.684
Kurtosis:                       2.897   Cond. No.                         628.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## (b) Interpretation of coefficients

- **Price:** Holding Urban and US constant, a one-unit increase in Price is associated with an average decrease of about 0.0545 in Sales.
- **Urban[T.Yes]:** Holding Price and US constant, Urban stores have about 0.022 lower Sales than non-Urban stores on average. This effect is very small and not statistically significant.
- **US[T.Yes]:** Holding Price and Urban constant, stores located in the US have about 1.20 higher Sales than stores outside the US on average.


## (c) Write out the model in equation form, being careful to handle the qualitative variables properly.


The variable Urban is a qualitative variable with values "Yes" and "No".
In the regression model, Urban[T.Yes] equals 1 for Urban = "Yes" and 0 otherwise.
Similarly, US[T.Yes] equals 1 for US = "Yes" and 0 otherwise.

Predicted Sales = 13.0435 − 0.0545 × Price − 0.0219 × Urban[T.Yes] + 1.2006 × US[T.Yes]


## (d) For which of the predictors can you reject the null hypothesis H0 : βj = 0?

Based on the p values from the regression output, we reject the null hypothesis for Price and US because their p values are very small. 
We fail to reject the null hypothesis for Urban since its p value is large.


##  On the basis of your response to the previous question, fit a smaller model that only uses the predictors for which there is evidence of association with the outcome

Based on the hypothesis testing results, Urban does not show evidence of association with Sales. Therefore, we fit a smaller model using only Price and US as predictors.


In [9]:
# Fit reduced model using only significant predictors
model_reduced = smf.ols("Sales ~ Price + US", data=carseats).fit()
model_reduced.summary()


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Sales   R-squared:                       0.239
Model:                            OLS   Adj. R-squared:                  0.235
Method:                 Least Squares   F-statistic:                     62.43
Date:                Sun, 21 Dec 2025   Prob (F-statistic):           2.66e-24
Time:                        10:11:15   Log-Likelihood:                -927.66
No. Observations:                 400   AIC:                             1861.
Df Residuals:                     397   BIC:                             1873.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     13.0308      0.631     20.652      0.000      11.790      14.271
US[T.Yes]      1.1996      0.258      4.641      0.000       0.692       1.708
Price         -0.0545      0.005    -10.416      0.000      -0.065      -0.044
==============================================================================
Omnibus:                        0.666   Durbin-Watson:                   1.912
Prob(Omnibus):                  0.717   Jarque-Bera (JB):                0.749
Skew:                           0.092   Prob(JB):                        0.688
Kurtosis:                       2.895   Cond. No.                         607.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## (f) How well do the models in (a) and (e) fit the data?

The full model has an R squared value of about 0.239, while the reduced model has an R squared value of about 0.239 as well. 
The adjusted R squared values are also nearly identical. This shows that removing Urban does not meaningfully reduce the model fit, so the reduced model fits the data just as well as the full model.


## Is there evidence of outliers or high leverage observations in the model from (e)?

To assess outliers and high leverage observations, diagnostic measures such as residuals, leverage values, and Cook's distance can be examined. There is no strong evidence that the model results are driven by a single extreme outlier or highly influential observation. Overall, the reduced model appears to be reasonably stable.
